### DISCUSSION TITLE
Toggling Plane Widgets in Subplots

### DISCUSSION THREAD

**Discussion by HarryvL** [pyvista/pyvista#5936] [Q&A]
### First check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the PyVista documentation, with the integrated search.
- [X] I already searched in Google "How to X in PyVista" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I will NOT make duplicate discussions or issues in the future.

### Commit to Help

- [X] I commit to help with one of those options.

### Sample Code What is the problem, question, or error?
Write a short description telling me what you are doing, what you expect to happen, and what is currently happening.

```python
p = pv.Plotter(shape=(2, 2))

p.subplot(0, 0)
p.add_mesh_clip_plane(mesh1)

p.subplot(0, 1)
p.add_mesh_clip_plane(mesh2)

p.subplot(1, 0)
p.add_mesh_clip_plane(mesh3)

p.subplot(1, 1)
p.add_mesh_clip_plane(mesh4)

def toggle_plane(p):
    for widget in plane_widgets:
    widget.SetEnabled(not widget.GetEnabled())

p.add_key_event("t", lambda: toggle_plane(p))

p.show()
```

### Description

I want to toggle visibility of the clip plane widgets in 4 subplots by pressing a key ("t" in this case). Upon first press, all widgets become invisible (which is good), but when pressing again, all 4 widgets get added to the single subplot that has the focus, while they still control the clipping of all subplots from the one subplot that has the focus. What I want is that the 4 plane widgets get added to the 4 separate subplots they were in at the start.

### System Information

```shell
--------------------------------------------------------------------------------
  Date: wo apr 17 11:51:44 2024 CEST

                OS : Linux
            CPU(s) : 16
           Machine : x86_64
      Architecture : 64bit
       Environment : Python
        GPU Vendor : Intel
      GPU Renderer : Mesa Intel(R) UHD Graphics (TGL GT1)
       GPU Version : 4.6 (Core Profile) Mesa 21.2.6
  MathText Support : False

  Python 3.11.5 | packaged by conda-forge | (main, Aug 27 2023, 03:53:17) [GCC
  12.3.0]

           pyvista : 0.43.5
               vtk : 9.2.5
             numpy : 1.24.4
        matplotlib : 3.8.4
            scooby : 0.9.2
             pooch : 1.8.1
            pillow : 9.4.0
           imageio : 2.34.0
         pyvistaqt : 0.11.0
             PyQt5 : 5.15.9
             scipy : 1.11.3
            meshio : 5.3.5
--------------------------------------------------------------------------------
```

### Screenshots

![Lug_FaceLoad_Example](https://github.com/pyvista/pyvista/assets/35259498/61eddf82-6288-4c54-ae2a-3585ab286776)

**Comment by banesullivan** [comment]
Ah, unfortunately, this looks to be a limitation of the current implementation and we likely need to refactor the widget helper methods to leverage a subplot context manager like `poked_subplot` which I implemented for the picking utilities in #4254.
